### DISCUSSION TITLE
How to save gTTS audio response in a temporary file ?

### DISCUSSION THREAD

**Discussion by Morojs** [pndurette/gTTS#284] [General]
i am just wondering if it is possible to save gTTS audio response in a temporary file with python ? 

this example doesn't work for me
```
from gtts import gTTS
import tempfile
from playsound import playsound

tempWavFile = tempfile.TemporaryFile(suffix="wav")

tts=gTTS('hello , how can i help you')
tts.save(tempfile)
```

**Comment by pndurette** [comment]
Hi! Two things I see here:

1. You're actually passing the `tempfile` _module_ to `gTTS.save()`, I think you meant to pass `tempWavFile`?
(also, by the way, it's `mp3` format that is generated)

2. And second, `tempfile.TemporaryFile()` returns a file-like object, and `gTTS.save()` is for files. What you want is [`gTTS.write_to_fp()`](https://gtts.readthedocs.io/en/latest/module.html#gtts.tts.gTTS.write_to_fp) instead!

**Reply by Morojs** [reply]
it works with gTTS.write_to_fp() , I need to play this wave file "tempWavFile" with playsound pkg like the code below :

````
tempWavFile = tempfile.TemporaryFile(suffix="wav")
tts=gTTS('hello , how can i help you')
tts.write_to_fp(tempWavFile) 
playsound(tempWavFile) 
````
#traceback
```
Traceback (most recent call last):
  File "D:\workspace\current-workspace\github\win-bot\test.py", line 10, in <module>
    playsound(tempWavFile)
  File "C:\Users\hp\AppData\Local\Programs\Python\Python39\lib\site-packages\playsound.py", line 35, in _playsoundWin
    winCommand('open "' + sound + '" alias', alias)
TypeError: can only concatenate str (not "_TemporaryFileWrapper") to str
```

i know that the tempWavFile it's not compatible with gTTS.write_to_fp() arg type  , is there any way to make it readable for gTTS.write_to_fp() ?
